dietpi/dietpi-software

```diff
@@ -11723,43 +11746,14 @@ _EOF_
 			# On fresh install, pre-configure for Sonarr and Radarr when found
 			if [[ ! -f '/mnt/dietpi_userdata/bazarr/config/config.ini' ]] && (( ${aSOFTWARE_INSTALL_STATE[144]} + ${aSOFTWARE_INSTALL_STATE[145]} > 0 ))
 			then
-				local timeout=15
-				(( $G_HW_ARCH > 2 )) || timeout=20
-				G_EXEC_DESC='Starting Bazarr to pre-create config file and pre-configure Sonarr and/or Radarr connection' G_EXEC systemctl start bazarr
-				local i=0
-				until [[ -f '/mnt/dietpi_userdata/bazarr/config/config.ini' || $i -ge $timeout ]]
-				do
-					((i++))
-					G_DIETPI-NOTIFY -2 "Waiting for Bazarr config file to be created ($i/$timeout)"
-					sleep 1
-				done
-				sleep 1
-				G_EXEC_NOHALT=1 G_EXEC systemctl stop bazarr
-				sleep 1
-				if [[ ! -f '/mnt/dietpi_userdata/bazarr/config/config.ini' ]]
-				then
-					G_DIETPI-NOTIFY 1 'Waiting for Bazarr config file to be created timed out, skipping Sonarr and Radarr pre-configuration'
+				CONFIG_CREATE /mnt/dietpi_userdata/bazarr/config/config.ini bazarr
 				else
 					# Sonarr
 					if (( ${aSOFTWARE_INSTALL_STATE[144]} > 0 ))
 					then
 						if [[ ! -f '/mnt/dietpi_userdata/sonarr/config.xml' ]]
 						then
-							G_EXEC_DESC='Starting Sonarr to pre-create config file' G_EXEC systemctl start sonarr
-							i=0
-							until [[ -f '/mnt/dietpi_userdata/sonarr/config.xml' ]] && grep -q ApiKey /mnt/dietpi_userdata/sonarr/config.xml || (( $i >= $timeout ))
-							do
-								((i++))
-								G_DIETPI-NOTIFY -2 "Waiting for Sonarr config file to be created ($i/$timeout)"
-								sleep 1
-							done
-						fi
-						sleep 1
-						G_EXEC_NOHALT=1 G_EXEC systemctl stop sonarr
-						sleep 1
-						if [[ ! -f '/mnt/dietpi_userdata/sonarr/config.xml' ]]
-						then
-							G_DIETPI-NOTIFY 1 'Sonarr config file creation timed out, skipping Sonarr pre-configuration'
+							CONFIG_CREATE /mnt/dietpi_userdata/sonarr/config.xml sonarr
```

Comment:
The conditional structure does not work anymore like this:
If the file already exist **or** has been successfully pre-created, the below pre-configuration shall be done.
So it could be:
```
if [[ -f '/mnt/dietpi_userdata/sonarr/config.xml' ]] || CONFIG_CREATE /mnt/dietpi_userdata/sonarr/config.xml sonarr
then
```

The is another thing:
- The file does not only need to be just created, but the `ApiKey` needs to be present, which is done after a few seconds.
- We need to add an optional content check as well, e.g. via variable ($3 would be optional as timeout): `CONFIG_CREATE_CONTENT='ApiKey'`
- Then in the function the check could be:
  ```
  local content=$CONFIG_CREATE_CONTENT
  unset -v CONFIG_CREATE_CONTENT
  ...
  until [[ -f $file ]] && { [[ ! $content]] || grep -q "$content" "$file"; } || [[ $i -ge $timeout ]] 
  ...
  [[ -f $file ]] && { [[ ! $content]] || grep -q "$content" "$file"; } && return 0

  G_DIETPI-NOTIFY 1 "Waiting for {$service^} config file to be created timed out, skipping pre-configuration"
  return 1
  ```